app.rb

```diff
@@ -1,34 +1,171 @@
-
-p "books rented by #{person.name}:"
-person.rentals.each { |ren| p [ren.date, ren.book.title, ren.book.author] }
-p "books rented by #{person2.name}:"
-person2.rentals.each { |ren| p [ren.date, ren.book.title, ren.book.author] }
-p "books rented by #{person3.name}:"
-person3.rentals.each { |ren| p [ren.date, ren.book.title, ren.book.author] }
-p "people who rented #{book1.title} by #{book1.author}:"
-book1.rentals.each { |ren| p [ren.date, ren.person.name] }
-p "people who rented #{book2.title} by #{book2.author}:"
-book2.rentals.each { |ren| p [ren.date, ren.person.name] }
+
+class App
+  def initialize
+    @books = []
+    @persons = []
+    @rentals = []
+  end
+
+  def create_teacher
+    puts '-creating a new teacher-'
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'specialization: '
+    spec = gets.chomp
+    @persons.push(Teacher.new(age, spec, name))
+    puts "Person created successfully\n"
+  end
+
+  def permission?
+    print "Has parents' permission? [Y/N]: "
+    option = gets.chomp.capitalize
+    case option
+    when 'N'
+      false
+    when 'Y'
+      true
+    else
+      permission?
+    end
+  end
+
+  def create_student
+    puts '-creating a new student-'
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp.to_i
+    permission = true
+    permission = permission? if age < 18
+    @persons.push(Student.new(age, name, parent_permission: permission))
+    puts "Person created successfully\n"
+  end
+
+  def create_person
+    print "\Do you want to create a student (1) or a teacher (2)? [Input the number]: "
+    option = gets.chomp.to_i
+    case option
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      create_person
+    end
+  end
+
+  def create_book
+    puts '-creating new book-'
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    @books.push(Book.new(title, author))
+  end
+
+  def select_book
+    puts 'Select a book from the list by [number]:'
+    @books.each_with_index { |book, i| puts "[#{i}] #{book.title} by #{book.author}" }
+    print 'Enter [number]: '
+    index = gets.chomp.to_i
+    return @books[index] if @books[index]
+
+    select_book
+  end
+
+  def select_person
+    puts 'Select a person from the list by [number]:'
+    @persons.each_with_index { |person, i| puts "[#{i}] [#{person.class.name}] ID: #{person.id} Name: #{person.name}" }
+    print 'Enter [number]: '
+    index = gets.chomp.to_i
+    return @persons[index] if @persons[index]
+
+    select_person
+  end
+
+  def create_rental
+    book = select_book
+    person = select_person
+    print 'insert date: '
+    date = gets.chomp
+    @rentals.push(Rental.new(date, book, person))
+    puts "rental created successfully.\n"
+  end
+
+  def all_books
+    @books.each_with_index { |book, i| puts "#{i + 1} - #{book.title} by #{book.author}" }
+  end
```

Comment:
- [ ] On listing the books there is an extra line that shouldn't be there, please 🙏🏽 correct this behavior to match with the original design 👇🏽 

![image](https://user-images.githubusercontent.com/58642949/226491203-c917b147-249d-443c-b268-f8752f6a51a4.png)